scripts/common.py

```diff
@@ -712,3 +715,98 @@ def __enter__(self):
 
     def __exit__(self, etype, value, traceback):
         os.chdir(self.saved_path)
+
+
+@contextlib.contextmanager
+def managed_process(args, shell=False, timeout=60, **kwargs):
+    """Context manager for starting and stopping a process gracefully.
+
+    Args:
+        args: list(*). A sequence of program arguments, where the program to
+            execute is the first item. All items are converted to strings first.
+        shell: bool. Whether the command should be run inside of its own shell.
+        timeout: int. The time allotted for the managed process and its
+            descendants to terminate themselves, in seconds. After the timeout,
+            any remaining processes will be killed abruptly.
+        **kwargs: dict(str: *). Same kwargs as `subprocess.Popen`.
+
+    Yields:
+        psutil.Process. The process managed by the context manager.
+    """
+    if PSUTIL_DIR not in sys.path:
+        sys.path.insert(1, PSUTIL_DIR)
+    import psutil
+
+    stripped_args = (python_utils.UNICODE(arg).strip() for arg in args)
+    non_empty_args = (arg for arg in stripped_args if arg)
+    popen_args = ' '.join(non_empty_args) if shell else list(non_empty_args)
+
+    root_proc = psutil.Popen(popen_args, shell=shell, **kwargs)
```

Comment:
I actually find process/processes to be less comprehensible, since the mess of `s`'s at the end makes it hard to tell what's plural and what's not.

I think we should follow `psutil`'s convention and leave it as `proc`.

WDYT?

Reply:
Resolved offline, it's OK to keep.